ledger-tool/src/main.rs

```diff
@@ -899,6 +899,80 @@ fn open_blockstore(
+        cf::Root::NAME => Some(cf::Root::slot(cf::Root::index(key))),
+        cf::Index::NAME => Some(cf::Index::slot(cf::Index::index(key))),
+        cf::ShredData::NAME => Some(cf::ShredData::slot(cf::ShredData::index(key))),
+        cf::ShredCode::NAME => Some(cf::ShredCode::slot(cf::ShredCode::index(key))),
+        cf::TransactionStatus::NAME => Some(cf::TransactionStatus::slot(
+            cf::TransactionStatus::index(key),
+        )),
+        cf::AddressSignatures::NAME => Some(cf::AddressSignatures::slot(
+            cf::AddressSignatures::index(key),
+        )),
+        cf::TransactionMemos::NAME => None, // does not implement slot()
+        cf::TransactionStatusIndex::NAME => None, // does not implement slot()
+        cf::Rewards::NAME => Some(cf::Rewards::slot(cf::Rewards::index(key))),
+        cf::Blocktime::NAME => Some(cf::Blocktime::slot(cf::Blocktime::index(key))),
+        cf::PerfSamples::NAME => Some(cf::PerfSamples::slot(cf::PerfSamples::index(key))),
+        cf::BlockHeight::NAME => Some(cf::BlockHeight::slot(cf::BlockHeight::index(key))),
+        cf::ProgramCosts::NAME => None, // does not implement slot()
+        cf::OptimisticSlots::NAME => {
+            Some(cf::OptimisticSlots::slot(cf::OptimisticSlots::index(key)))
+        }
+        &_ => None,
+    }
+}
+
+fn print_blockstore_file_info(
+    blockstore: &Blockstore,
+    file_name: &Option<String>,
+) -> Result<(), String> {
+    match blockstore.live_files_metadata() {
+        Ok(live_files) => {
+            // All files under live_files_metadata are prefixed with "/".
+            let sst_file_name = file_name.as_ref().map(|name| format!("/{}", name));
+            for file in live_files {
+                if sst_file_name.is_none() || file.name.eq(sst_file_name.as_ref().unwrap()) {
+                    println!(
+                        "[{}] cf_name: {}, level: {}, start_slot: {:?}, end_slot: {:?}, size: {}, num_entries: {}",
+                        file.name,
+                        file.column_family_name,
+                        file.level,
+                        raw_key_to_slot(&file.start_key.unwrap(), &file.column_family_name),
+                        raw_key_to_slot(&file.end_key.unwrap(), &file.column_family_name),
+                        file.size,
+                        file.num_entries,
+                    );
+                    if sst_file_name.is_some() {
+                        return Ok(());
+                    }
+                }
+            }
+            if sst_file_name.is_some() {
+                return Err(format!(
+                    "Failed to find or load the metadata of the specified file {:?}",
+                    file_name
+                ));
+            }
+        }
+        Err(err) => {
+            return Err(format!(
+                "Unable to identify slot range of the specified file {:?}: {:?}",
+                file_name, err
+            ));
+        }
```

Comment:
Just incase you shift it around, the error text is currently inaccurate as this corresponds to when `live_files_metadata()` returns an error.

However, if you end up agreeing with my comment to use `?`, I think this text / handling can go away altogether